napari/utils/key_bindings.py

```diff
@@ -182,6 +186,284 @@ def normalize_key_combo(key_combo):
+                # future.
+                action.command(*args)
+                for b in self._buttons.get(name, []):
+                    b._animation.start()
+
+            keymappable.bind_key(sht)(flash)
+
+    def bind_button(self, name, button):
+        """
+        Bind `button` to trigger Action `name` on click.
+        """
+        action = self._actions[name]
+
+        button.clicked.connect(
+            lambda: call_with_context(action.command, self.context)
+        )
+        button.destroyed.connect(lambda: self._buttons[name].remove(button))
+
+        self._buttons[name].append(button)
+        self._update_gui_elements(name)
+
+    def bind_shortcut(self, name, shortcut):
+        """
+        bind shortcut `shortcut` to trigger action `name`
+        """
+        self._shortcuts[name] = shortcut
+        self._update_shortcut_bindings(name)
+        self._update_gui_elements(name)
+
+    def bind_qaction(self, name, qaction):
+        """
+        Bind the given qaction to an action.
+
+        This will also update the description and shortcut when those changes.
+
+        Same as for shortcut; make ui element flash when action is triggered.
+        """
+        self._qactions[name] = qaction
+        action = self._actions[name]
+
+        def flash(*args):
+            call_with_context(action.command, self.context)
+
+            for b in self._buttons.get(name, []):
+                b._animation.start()
+
+        qaction.destroyed.connect(lambda: self._qactions.pop(name, None))
+        qaction.triggered.connect(flash)
+        self._update_gui_elements(name)
+
+    def unbind_shortcut(self, name):
+        """
+        unbind shortcut for action name
+        """
+        action = self._actions[name]
+        sht = self._shortcuts.get(name)
+        if hasattr(action.keymappable, 'bind_key'):
+            action.keymappable.bind_key(sht)(None)
+        del self._shortcuts[name]
+        self._update_gui_elements(name)
+
+
+action_manager = ActionManager()
+
+
+_actions = {
+    'V': 'toggle_selected_visibility',
+    'Control-G': 'toggle_grid',
+    'Shift-Down': 'also_select_layer_below',
+    'Shift-Up': 'also_select_layer_above',
+    'Down': 'select_layer_below',
+    'Up': 'select_layer_above',
+    'Control-A': 'select_all',
+    'Control-T': 'transpose_axes',
+    'Control-R': 'reset_view',
+    'Control-Shift-C': 'toggle_console_visibility',
+    'Control-E': 'roll_axes',
+    'Alt-Down': 'focus_axes_down',
+    'Alt-Up': 'focus_axes_up',
+    'Right': 'increment_dims_right',
+    'Left': 'increment_dims_left',
+    'Control-Y': 'toggle_ndisplay',
+    'Control-Shift-Backspace': 'remove_all_layers',
+    'Control-Shift-Delete': 'remove_all_layers',
+    'Control-Backspace': 'remove_selected',
+    'Control-Delete': 'remove_selected',
+}
+
+for (shortcut, action_name) in _actions.items():
+    action_manager.bind_shortcut(action_name, shortcut)
```

Comment:
I'm not sure where the best place to do this is, especially WRT configurability.

Reply:
On the viewer/main window, we should probably load shortcuts and do the binding?

Reply:
see my other comment... it's not immediately clear to me why this dict needs to exist, and why it cant be part of the call to `register` (actually tied to the function itself)

Reply:
I did it this way for a few reason. 

 1) you may want to register an action without giving it a shortcut. 
 2) it makes it _much_ easier to have traduction and configurability even if you haven't loaded the plugins.



```
default_shortcut = {...}
default_shortcuts.update(get_localized_shortcut())
default_shortcuts.update(get_preference_shortcut())

for key, action in ...
```
If you don't do that; then you need this convoluted logic:
  - If someone register a new action:
     - does it have a default shortcut XXX ?
     - does XXX overwrite a user defined shortcut ?
     - Does the user want to replace XXX with YYY ?
```
It will also triggers a lot of ordering problem, and not make it easy to find all the defauld shortcuts.


I think in the core of the logic we want to keep the register and bind-shortcut separate, and  we can provide a decorator that allow to pass a default shortcut when you register an action.